### PR TITLE
ENH add MPI

### LINKED ISSUE #51: MPI support
It is more a question than a proper issue for this feedstock.

I'd need fftw3_mpi in conda-forge. Similar to these Debian package:

- https://packages.debian.org/stretch/libfftw3-mpi
- https://packages.debian.org/stretch/libfftw3-mpi-dev

These packages depend on libfftw3 and openmpi. They do not contain the libfftw3 libraries.

It would be nice to have the same structure on conda, i.e. that the conda packages fftw and fftw_mpi would be compatible (fftw would be a dependence of fftw_mpi).

The code for fftwmpi-feedstock would be very similar than the one for this feedstock and it would also produce the same libraries as this recipe, so I'm wondering if it's a good thing to have another feedstock or if it would be possible with one feedstock to create two packages.

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #51

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin, rerender

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 1]
```diff
@@ -10,8 +10,14 @@ jobs:
   strategy:
     maxParallel: 8
     matrix:
-      linux_:
-        CONFIG: linux_
+      linux_mpimpich:
+        CONFIG: linux_mpimpich
+        UPLOAD_PACKAGES: True
+      linux_mpinompi:
```
Comment (rainwoodman) [inline .azure-pipelines/azure-pipelines-linux.yml]: linux_nompi seems to be less cluttered, if there is no strong reason to prefix every element with linux_mpi?

Comment (beckermr) [reply .azure-pipelines/azure-pipelines-linux.yml]: This is an internal build file and has no relevance for the final namespace.

**Comment by beckermr** [comment]
See the docs here @rainwoodman: http://conda-forge.org/docs/maintainer/knowledge_base.html#message-passing-interface-mpi

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 1 above)
Comment (beckermr) [reply .azure-pipelines/azure-pipelines-linux.yml]: Also this file is autogenerated by the conda-forge tooling so we absolutely cannot change it.

**Comment by beckermr** [comment]
OK @conda-forge/fftw @paugier @grlee77 @rainwoodman, this PR is ready to go. I'd be good for someone to test an actual build against the package to make sure everything is in order. I don't have anything handy.

**Comment by beckermr** [comment]
Mmmk. Merging.

**Comment by grlee77** [comment]
Thanks @beckermr, this is great! Sorry I didn't get a chance to look at this sooner. 

The implementation looks in line with the recommendation in the conda-forge docs and I can see the expected MPI variants for OS X and Linux showing up at https://anaconda.org/conda-forge/fftw/files. 

I also verified that the build number priority is giving `nompi` as the default if I request just `conda install fftw`.

**Pull request merged**